src/udp_listener.rs

```diff
@@ -94,26 +95,30 @@ impl RaiiUdpListener {
            stop_flag: Arc<AtomicBool>,
            connection_map: Arc<Mutex<HashMap<PeerId, Vec<Connection>>>>,
            mc: Arc<MappingContext>) {
-        let mut read_buf = [0; 1024];
-
-        while !stop_flag.load(Ordering::SeqCst) {
-            if let Ok((bytes_read, peer_addr)) = udp_socket.recv_from(&mut read_buf) {
-                if let Ok(msg) = deserialise::<ListenerRequest>(&read_buf[..bytes_read]) {
-                    let our_public_key = our_public_key.clone();
-                    let event_tx = event_tx.clone();
-                    let connection_map = connection_map.clone();
-                    let mc = mc.clone();
-                    let _  = thread!("Bootstrap uTP accept", move || {
-                        RaiiUdpListener::handle_request(msg,
-                                                        our_public_key,
-                                                        peer_addr,
-                                                        event_tx,
-                                                        connection_map,
-                                                        mc);
-                    });
+
+        let udp_socket = &udp_socket;
+        crossbeam::scope(move |scope| {
```

Comment:
I'm not sure I'm happy in creating yet another thread just to do a `udp_socket.clone()`. What do you think about [this alternative approach](https://github.com/maidsafe/crust/pull/638)?